Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM golang:1.17
+
+WORKDIR /
+
+COPY api api
+COPY cmd cmd
+COPY pkg pkg
+COPY test test
+COPY main.go main.go
+COPY Makefile Makefile
+COPY tools.go tools.go
+COPY go.mod go.mod
+COPY go.sum go.sum
```

Comment:
Should we move this further up in the Dockerfile so we can utilize already cached layers and don't have to re-download the module cache anytime we make a code change (that doesn't affect dependencies):

```dockerfile
COPY go.mod go.mod
COPY go.sum go.sum
RUN go mod download

COPY api api
...
```

Any thoughts?

Reply:
Ahh, good idea. I like this.